python-flask/app/simulator/routes.py

```diff
@@ -0,0 +1,125 @@
+from flask import Blueprint
+from flask import jsonify, request, g, abort
+from werkzeug.security import generate_password_hash
+from datetime import datetime
+from app.models.user import User, Follower
+from app.models.message import Message
+from app.models.latest import Latest
+from app.utils import not_req_from_simulator, update_latest, get_user_id
+from app.extensions import db
+
+sim_bp = Blueprint('sim', __name__)
+
+@sim_bp.route("/latest", methods=["GET"])
+def get_latest():
+    latest_entry = Latest.query.first()  # Fetch the first and only entry
+    latest_processed_command_id = latest_entry.value if latest_entry else -1
+    return jsonify({"latest": latest_processed_command_id})
+
+@sim_bp.route("/register", methods=["POST"])
+def register():
+    update_latest(request)
+    not_from_sim_response = not_req_from_simulator(request)
+    if not_from_sim_response:
+        return not_from_sim_response
+    request_data = request.json
+    error = None
+    if not request_data.get("username"):
+        error = "You have to enter a username"
+    elif not request_data.get("email") or "@" not in request_data["email"]:
+        error = "You have to enter a valid email address"
+    elif not request_data.get("pwd"):
+        error = "You have to enter a password"
+    elif get_user_id(request_data["username"]) is not None:
+        error = "The username is already taken"
+    else:
+        user = User(username=request_data["username"], email=request_data["email"], pw_hash=generate_password_hash(request_data["pwd"]))
+        db.session.add(user)
+        db.session.commit()
+
+    if error:
+        return jsonify({"status": 400, "error_msg": error}), 400
+    else:
+        return "", 204
+
+
+@sim_bp.route("/msgs", methods=["GET"])
+def messages():
+    """Return all latest messages."""
+    update_latest(request)
+    not_from_sim_response = not_req_from_simulator(request)
+    if not_from_sim_response:
+        return not_from_sim_response
+
+    no_msgs = request.args.get("no", type=int, default=100)
+    messages = db.session.query(Message, User).join(User, Message.author_id == User.user_id) \
+        .filter(Message.flagged == 0) \
```

Comment:
We need align on whether or not we keep the 'flagged' column in messages

Reply:
True, lets discuss this on Friday